datafusion/physical-expr/src/functions.rs

```diff
@@ -266,16 +288,20 @@ where
                 ColumnarValue::Array(a) => Some(a.len()),
             });
 
-        // to array
-        let args = if let Some(len) = len {
-            args.iter()
-                .map(|arg| arg.clone().into_array(len))
-                .collect::<Vec<ArrayRef>>()
-        } else {
-            args.iter()
-                .map(|arg| arg.clone().into_array(1))
-                .collect::<Vec<ArrayRef>>()
-        };
+        let inferred_length = len.unwrap_or(1);
+        let args = args
```

Comment:
Not sure how functional you want, but you can probably use `.zip(hints.iter())` here and make it more suscinct

You may have to extend hints too with `chain(std::iter::repeat(false))` or something to ensure all args are processed